### PR TITLE
[Justice Counts] Staging Badge FE: add badge to indicate environment

### LINKED ISSUE #519: Badge to indicate whether we're on staging
See Slack thread [here](https://recidiviz.slack.com/archives/C02VDN1LSSW/p1679519918416769) for more context.

I'm not sure if the FE currently has a way of knowing whether we are on staging or not, but if not, we could easily start passing this from the BE to FE in our `/init` API call. The BE can determine this from the `in_gcp_staging` method in `recidiviz.utils.environment` and then pass that value over as a key in the JSON response.

I'm assigning this to @morden35 since it could be a nice lightweight FE ramp up task, but cc'ing @mxosman to help!

### PR DESCRIPTION
**Pull request by morden35**
## Description of the change

This PR adds a 'Local' or 'Staging' badge on the header bar depending on which environment the app is deployed on.

Screenshot 2023-04-12 at 11 03 08 AM

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Related to [#20126](https://github.com/Recidiviz/recidiviz-data/pull/20126)
Closes #519

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by morden35** [comment]
Open to any and all feedback on the styling of the 'badge'! It might look better if it was imported as an image. Right now the text looks a bit squished to me.

File: publisher/src/components/Header/Header.tsx [hunk 1]
```diff
@@ -50,6 +51,12 @@ const Header = observer(() => {
       >
         <Logo src={logo} alt="" />
         <LogoName>Justice Counts</LogoName>
+        {rootStore.api.environment === "local" && (
```
Comment (mxosman) [inline publisher/src/components/Header/Header.tsx:54]: No need to access the API store via the rootStore - you can extract it from the `useStore()` function on L33 (which uses React's Context API to keep all of the stores/states in sync):

```
L33:       const { userStore, guidanceStore, api } = useStore();
```

Then you can just use `api.environment` here.

Comment (morden35) [reply publisher/src/components/Header/Header.tsx:54]: oh awesome! thank you!

File: publisher/src/components/Header/Header.styles.tsx [hunk 2]
```diff
@@ -67,3 +67,16 @@ export const LogoName = styled.div`
...
+  background-color: red;
+  color: white;
+  height: 64px;
+  width: 100%;
+  line-height: 64px;
+
+  @media only screen and (max-width: ${MIN_DESKTOP_WIDTH}px) {
+    display: none;
+  }
+`;
```
Comment (mxosman) [inline publisher/src/components/Header/Header.styles.tsx:82]: Ooo - I think you can use the `Badge` component instead! It'll take care of all of the styling and match the badges we currently have - all you need to do is provide it a color.

You can import it from the common repo like this:

`import { Badge } from "@justice-counts/common/components/Badge";`

And use it like this:
```
          // BadgeColor options currently: "RED" | "GREEN" | "ORANGE" | "GREY"
          <Badge
            color="COLOR" // you could also have a conditional here like color={condition ? "RED" : "GREEN"}
          >
            Your label goes here
          </Badge>
```

**Comment by mxosman** [comment]
This is so cool and will be super helpful, Michelle!!

File: publisher/src/components/Header/Header.styles.tsx (hunk 2 above)
Comment (morden35) [reply publisher/src/components/Header/Header.styles.tsx:82]: excellent! thank you!

**Review by lilidworkin — COMMENTED** [review]
This looks perfect to me @morden35 ! Will let @mxosman do the final review/accept!

**Review by mxosman — APPROVED** [review]
Looks fantastic, Michelle!!

**Comment by morden35** [comment]
> Looks fantastic, Michelle!!

Thank you!

**Pull request merged**